### DISCUSSION TITLE
fork/exec /snap/bin/chromium: permission denied

### DISCUSSION THREAD

**Discussion by ghost** [dstotijn/hetty#127] [General]
Hey there,

Running on Ubuntu. I have chrome installed & chromium. When I run hetty --chrome I get the following:

```
$ hetty --chrome
2024/02/25 16:58:02 INFO [main] Hetty (v0.7.0) is running on :8080 ...
2024/02/25 16:58:02 INFO [main] Get started at http://localhost:8080
2024/02/25 16:58:02 ERROR [main] Failed to navigate to http://localhost:8080. {"error": "fork/exec /snap/bin/chromium: permission denied"}
```

Any guidance on what is causing this error?

Thanks!

**Comment by dstotijn** [comment]
Hi Johah, although it's hard to tell from just these logs, I suspect that executing `/snap/bin/chromium` is being restricted on your Ubuntu installation by [AppArmor](https://apparmor.net/). If so, you might want to check your local AppArmor logs to confirm this (perhaps in the log directory mentioned [here](https://wiki.ubuntu.com/DebuggingApparmor#Debugging_procedure)).

**Reply by m0veax** [reply]
Hey, i ran into the same issue and can confirm that AppArmor blocks the request:

```
Oct 29 22:14:41 drunken-fiddler kernel: [ 4215.008807] audit: type=1400 audit(1730236481.103:239): apparmor="DENIED" operation="open" class="file" profile="snap.hetty.hetty" name="/etc/pop-os/os-release" pid=33466 comm="snap-exec" requested_mask="r" denied_mask="r" fsuid=1000 ouid=0
Oct 29 22:14:41 drunken-fiddler kernel: [ 4215.094095] audit: type=1400 audit(1730236481.188:240): apparmor="DENIED" operation="exec" class="file" profile="snap.hetty.hetty" name="/usr/bin/snap" pid=33497 comm="hetty" requested_mask="x" denied_mask="x" fsuid=1000 ouid=0
```
